src/_blog/go-libp2p-2022.md

```diff
@@ -0,0 +1,117 @@
+This resulted in a net saving one round trip, which may seem minimal but is big waste during connection establishment!
+
+<!-- To learn more about how early muxer negotiation works, please read: https://github.com/libp2p/docs/pull/274 -->
+<!-- above docs is a blocker for merge -->
+
+## Project Improvements 🏡
+
+### DoS Protection 🏰 & Resource Management 📦
+
+We added the [Resource Manager component](https://github.com/libp2p/go-libp2p/tree/master/p2p/host/resource-manager#readme) in [v0.18.0](https://github.com/libp2p/go-libp2p/releases/tag/v0.18.0). This enabled configuring limits to control the number of incoming & outgoing connections and streams, the number of streams per protocol & service, as well as configuring libp2p memory usage.
+
+Furthermore, go-libp2p also added autoscaling limits and canonical log lines in [v0.21.0](https://github.com/libp2p/go-libp2p/releases/tag/v0.21.0).
+All of these are key to DoS mitigation. Please read our documentation on [how you can incorporate DoS mitigation into libp2p](https://docs.libp2p.io/concepts/security/dos-mitigation/).
+
+### Interoperability Testing ⚙️
+
+libp2p implementations in different languages support a wide variety of modules (i.e., transport protocols, stream multiplexers, secure channels, etc.). Furthermore, due to the advent of multiple new transports (like WebTransport, WebRTC), we began a concerted effort to test interoperability between go-libp2p and implementations in other languages (like Rust, JS, and Nim.)
+
+The details of our interoperability testing efforts can be seen in the shared [libp2p/test-plans Roadmap](https://github.com/libp2p/test-plans/blob/master/ROADMAP.md#a-multi-dimensional-testinginterop-visibility).
+
+### Monorepo Consolidation 1️⃣
+
+go-libp2p is a monorepo as of the [v0.22.0 release](https://github.com/libp2p/go-libp2p/releases/tag/v0.22.0).
+
+This effort was initiated to address repo sprawl and consolidate go-libp2p modules; the result was a monorepo and a big quality of life improvement. The go-libp2p maintainers always wanted to address sprawl but it wasn’t possible until started after [lazy module loading was added to Go 1.17](https://go.dev/ref/mod#lazy-loading).
+The consolidation started in [v0.18.0](https://github.com/libp2p/go-libp2p/releases/tag/v0.18.0) and finished in v0.22.0 release (where go-libp2p-core was finally migrated.)
+
+### Simplify libp2p Setup Logic with Fx 🦄
+
+In [v0.24.0](https://github.com/libp2p/go-libp2p/releases/tag/v0.24.0) go-libp2p included a change to use [Fx](https://github.com/uber-go/fx), a Go dependency injection library.
```

Comment:
Not really, this is not an outwards-facing change.

It just massively simplified how we construct our transports, and it will enable us to properly start and stop libp2p services in the future (see https://github.com/libp2p/go-libp2p/issues/1993).